File: backend/agents/architect.py
import os
from emergentintegrations.llm.chat import LlmChat, UserMessage
from datetime import datetime, timezone

class ArchitectAgent:
    def __init__(self, db, manager):
        self.db = db
        self.manager = manager
        self.llm = LlmChat(
            api_key=os.environ['EMERGENT_LLM_KEY'],
            session_id="architect",
            system_message="You are a software architect. Design system architecture including data models, API endpoints, file structure, and technology choices. Be specific and detailed."
        ).with_model("anthropic", "claude-3-7-sonnet-20250219")

    async def design(self, task_id: str, plan: str):
        await self._log(task_id, "Architect", "🏗️ Designing system architecture...")
        
        user_message = UserMessage(
            text=f"Based on this plan: {plan}\n\nCreate a detailed architecture design including: data models, API endpoints, file structure, component hierarchy."
        )
        
        try:
            response = await self.llm.send_message(user_message)
            await self._log(task_id, "Architect", f"✅ Architecture designed: {len(response)} chars")
            return {"architecture": response, "status": "success"}
        except Exception as e:
            await self._log(task_id, "Architect", f"❌ Architecture design failed: {str(e)}")
            return {"architecture": "", "status": "failed", "error": str(e)}

    async def _log(self, task_id: str, agent_name: str, message: str):
        log_doc = {
            "task_id": task_id,
            "agent_name": agent_name,
            "message": message,
            "timestamp": datetime.now(timezone.utc).isoformat()
        }
        await self.db.agent_logs.insert_one(log_doc)
        await self.manager.send_log(task_id, log_doc)

File: backend-go/internal/agents/architect.go
package agents

import (
	"context"
	"time"

	"github.com/catalyst/backend/internal/config"
	"github.com/catalyst/backend/internal/database"
	"github.com/catalyst/backend/internal/llm"
	"github.com/catalyst/backend/internal/models"
	"github.com/catalyst/backend/internal/websocket"
	"github.com/google/uuid"
)

type ArchitectAgent struct {
	db        *database.Database
	wsManager *websocket.Manager
	llmClient *llm.Client
}

func NewArchitectAgent(db *database.Database, wsManager *websocket.Manager, cfg *config.Config) *ArchitectAgent {
	return &ArchitectAgent{
		db:        db,
		wsManager: wsManager,
		llmClient: llm.NewClient(cfg.LLMKey, cfg.LLMProvider, cfg.LLMModel),
	}
}

func (a *ArchitectAgent) Design(taskID, plan string) (string, error) {
	a.log(taskID, "Architect", "🏗️ Designing system architecture...")

	systemPrompt := "You are a software architect. Design system architecture including data models, API endpoints, file structure, and technology choices. Be specific and detailed."
	userPrompt := "Based on this plan: " + plan + "\n\nCreate a detailed architecture design including: data models, API endpoints, file structure, component hierarchy."

	response, err := a.llmClient.SendMessage(systemPrompt, userPrompt)
	if err != nil {
		a.log(taskID, "Architect", "❌ Architecture design failed: "+err.Error())
		return "", err
	}

	a.log(taskID, "Architect", "✅ Architecture designed successfully")
	return response, nil
}

func (a *ArchitectAgent) log(taskID, agentName, message string) {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	log := models.AgentLog{
		ID:        uuid.New().String(),
		TaskID:    taskID,
		AgentName: agentName,
		Message:   message,
		Timestamp: time.Now(),
	}

	a.db.Logs.InsertOne(ctx, log)
	a.wsManager.SendLog(taskID, map[string]interface{}{
		"task_id":    taskID,
		"agent_name": agentName,
		"message":    message,
		"timestamp":  log.Timestamp.Format(time.RFC3339),
	})
}